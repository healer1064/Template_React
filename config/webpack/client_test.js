// @flow

const config = require('../../webpack.config')

it('webpack config matches snapshot', () => {

  expect(config).toMatchSnapshot()

})
