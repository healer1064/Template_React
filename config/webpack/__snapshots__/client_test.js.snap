// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack config matches snapshot 1`] = `
Array [
  Object {
    "devtool": "source-map",
    "entry": Object {
      "bundle": Array [
        "sanitize.css/sanitize.css",
        "./src/styles/fonts.css",
        "./src/styles/app.css",
        "./src/browser.index.js",
      ],
      "vendor": Array [
        "axios",
        "babel-polyfill",
        "classnames",
        "cookies-js",
        "debug",
        "immutable",
        "js-base64",
        "react",
        "react-dom",
        "react-helmet",
        "react-redux",
        "react-router",
        "react-router-redux",
        "redux",
        "redux-devtools-extension",
        "redux-form",
        "redux-observable",
        "redux-promise-middleware",
        "redux-thunk",
        "reselect",
        "rxjs",
        "rollbar-browser/dist/rollbar.umd.nojson.min.js",
        "transit-immutable-js",
      ],
    },
    "externals": undefined,
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "loader": "babel-loader?cacheDirectory=false",
          "test": /\\\\\\.js\\$/,
        },
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.css\\$/,
          "use": Array [
            Object {
              "loader": "/Users/artribou/dev/react-template/node_modules/extract-text-webpack-plugin/loader.js",
              "options": Object {
                "omit": 0,
                "remove": true,
              },
            },
            Object {
              "loader": "css-loader?minimize&modules&camelCase&importLoaders=1",
            },
            Object {
              "loader": "postcss-loader",
            },
          ],
        },
        Object {
          "test": /node_modules\\.\\*\\\\\\.css\\$/,
          "use": Array [
            Object {
              "loader": "/Users/artribou/dev/react-template/node_modules/extract-text-webpack-plugin/loader.js",
              "options": Object {
                "omit": 0,
                "remove": true,
              },
            },
            Object {
              "loader": "css-loader?minimize&importLoaders=1",
            },
            Object {
              "loader": "postcss-loader",
            },
          ],
        },
        Object {
          "test": /\\\\\\.\\(eot\\|mp4\\|otf\\)\\$/,
          "use": "file-loader",
        },
        Object {
          "test": /\\\\\\.\\(gif\\|jpeg\\|jpg\\|png\\|svg\\)\\$/,
          "use": "url-loader?limit=10000",
        },
        Object {
          "test": /\\\\\\.\\(woff\\|woff2\\)\\$/,
          "use": "url-loader?limit=10000&mimetype=application/font-woff",
        },
        Object {
          "test": /\\\\\\.ttf\\$/,
          "use": "url-loader?limit=10000&mimetype=application/octet-stream",
        },
      ],
    },
    "node": undefined,
    "output": Object {
      "chunkFilename": "[name]-[chunkhash].js",
      "filename": "[name]-[chunkhash].js",
      "path": "/Users/artribou/dev/react-template/build/public",
      "publicPath": "/static/",
    },
    "plugins": Array [
      DefinePlugin {
        "definitions": Object {
          "process.env": Object {
            "NODE_ENV": "\\"production\\"",
          },
        },
      },
      Object {
        "apply": [Function],
      },
      CommonsChunkPlugin {
        "async": undefined,
        "children": undefined,
        "chunkNames": Array [
          "vendor",
        ],
        "filenameTemplate": undefined,
        "ident": "/Users/artribou/dev/react-template/node_modules/webpack/lib/optimize/CommonsChunkPlugin.js0",
        "minChunks": Infinity,
        "minSize": undefined,
        "selectedChunks": undefined,
      },
      ExtractTextPlugin {
        "filename": "styles.css",
        "id": 1,
        "options": Object {
          "allChunks": true,
        },
      },
      AssetsWebpackPlugin {
        "options": Object {
          "filename": "assets.json",
          "fullPath": true,
          "path": "/Users/artribou/dev/react-template/build",
          "prettyPrint": false,
          "processOutput": [Function],
          "update": false,
        },
        "writer": [Function],
      },
      FaviconsWebpackPlugin {
        "options": Object {
          "background": "#fff",
          "emitStats": false,
          "icons": Object {
            "android": true,
            "appleIcon": true,
            "appleStartup": true,
            "coast": false,
            "favicons": true,
            "firefox": true,
            "opengraph": false,
            "twitter": false,
            "windows": false,
            "yandex": false,
          },
          "inject": true,
          "logo": "static/images/logo@2x.png",
          "persistentCache": true,
          "prefix": "icons-[hash]/",
          "statsFilename": "iconstats-[hash].json",
        },
      },
      CompressionPlugin {
        "algorithm": [Function],
        "asset": "[path].gz[query]",
        "compressionOptions": Object {
          "chunkSize": undefined,
          "dictionary": undefined,
          "flush": undefined,
          "level": 9,
          "memLevel": undefined,
          "strategy": undefined,
          "windowBits": undefined,
        },
        "deleteOriginalAssets": false,
        "filename": false,
        "minRatio": 0.8,
        "test": /\\\\\\.\\(js\\|css\\|html\\|json\\|ico\\|map\\|xml\\|txt\\|svg\\|eot\\|otf\\|ttf\\|woff\\|woff2\\)\\$/,
        "threshold": 10240,
      },
      UglifyJsPlugin {
        "options": Object {
          "sourceMap": true,
        },
      },
      LoaderOptionsPlugin {
        "options": Object {
          "minimize": true,
          "test": Object {
            "test": [Function],
          },
        },
      },
      OfflinePlugin {
        "EXTERNALS_KEY": ":externals:",
        "REST_KEY": ":rest:",
        "__tests": Object {
          "ignoreRuntime": false,
          "noVersionDump": false,
          "swMetadataOnly": false,
        },
        "assets": null,
        "cacheMaps": Array [],
        "entryPrefix": "__offline_",
        "errors": Array [],
        "externals": null,
        "flags": Object {},
        "hash": null,
        "hashesMap": null,
        "loaders": Object {},
        "options": Object {
          "AppCache": false,
          "ServiceWorker": Object {
            "entry": "/Users/artribou/dev/react-template/node_modules/offline-plugin/empty-entry.js",
            "events": true,
            "navigateFallbackForRedirects": true,
            "output": "sw.js",
            "prefetchRequest": Object {
              "cache": undefined,
              "credentials": "omit",
              "headers": undefined,
              "mode": "cors",
            },
            "scope": null,
          },
          "__tests": Object {
            "ignoreRuntime": false,
            "noVersionDump": false,
            "swMetadataOnly": false,
          },
          "alwaysRevalidate": undefined,
          "autoUpdate": false,
          "cacheMaps": null,
          "caches": "all",
          "excludes": Array [
            "**/.*",
            "**/*.map",
            "robots.txt",
          ],
          "externals": Array [],
          "ignoreSearch": Array [
            "**",
          ],
          "preferOnline": undefined,
          "publicPath": "/",
          "relativePaths": ":relativePaths:",
          "responseStrategy": "cache-first",
          "rewrites": [Function],
          "updateStrategy": "changed",
          "version": "v3.4.0-[hash]",
        },
        "pluginVersion": "4.7.0",
        "publicPath": "/",
        "relativePaths": ":relativePaths:",
        "responseStrategy": "cache-first",
        "rewrite": [Function],
        "strategy": "changed",
        "tools": Object {
          "ServiceWorker": ServiceWorker {
            "CACHE_NAME": "webpack-offline",
            "ENTRY_NAME": "serviceworker",
            "SW_DATA_VAR": "__wpo",
            "basePath": null,
            "entry": "/Users/artribou/dev/react-template/node_modules/offline-plugin/empty-entry.js",
            "events": true,
            "location": null,
            "navigateFallbackForRedirects": true,
            "navigateFallbackURL": undefined,
            "output": "sw.js",
            "pathRewrite": null,
            "prefetchRequest": undefined,
            "publicPath": undefined,
            "scope": undefined,
          },
        },
        "warnings": Array [],
      },
    ],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
        ".web.js",
      ],
      "modules": Array [
        "/Users/artribou/dev/react-template",
        "node_modules",
      ],
    },
    "target": "web",
  },
  Object {
    "devtool": "source-map",
    "entry": Object {
      "server.js": Array [
        "./server/index.js",
      ],
      "views/Html.js": "./server/views/Html.js",
    },
    "externals": Array [
      [Function],
    ],
    "module": Object {
      "rules": Array [
        Object {
          "exclude": /node_modules/,
          "loader": "babel-loader?cacheDirectory=false",
          "test": /\\\\\\.js\\$/,
        },
        Object {
          "exclude": /node_modules/,
          "test": /\\\\\\.css\\$/,
          "use": "css-loader/locals?minimize&modules&camelCase",
        },
        Object {
          "test": /node_modules\\.\\*\\\\\\.css\\$/,
          "use": "css-loader/locals?minimize",
        },
        Object {
          "test": /\\\\\\.\\(eot\\|mp4\\|otf\\)\\$/,
          "use": "file-loader",
        },
        Object {
          "test": /\\\\\\.\\(gif\\|jpeg\\|jpg\\|png\\|svg\\)\\$/,
          "use": "url-loader?limit=10000",
        },
        Object {
          "test": /\\\\\\.\\(woff\\|woff2\\)\\$/,
          "use": "url-loader?limit=10000&mimetype=application/font-woff",
        },
        Object {
          "test": /\\\\\\.ttf\\$/,
          "use": "url-loader?limit=10000&mimetype=application/octet-stream",
        },
      ],
    },
    "node": Object {
      "__dirname": false,
      "__filename": false,
    },
    "output": Object {
      "filename": "[name]",
      "libraryTarget": "commonjs2",
      "path": "/Users/artribou/dev/react-template/build",
    },
    "plugins": Array [
      BannerPlugin {
        "banner": "require(\\"source-map-support\\").install();",
        "options": Object {
          "banner": "require(\\"source-map-support\\").install();",
          "entryOnly": false,
          "raw": true,
        },
      },
    ],
    "resolve": Object {
      "extensions": Array [
        ".js",
        ".json",
        ".web.js",
      ],
      "modules": Array [
        "/Users/artribou/dev/react-template",
        "node_modules",
      ],
    },
    "target": "node",
  },
]
`;
